inc/functions-icons.php

```diff
@@ -0,0 +1,206 @@
+<?php
+/**
+ * SVG icons related functions and filters.
+ *
+ * @package WordPress
+ * @subpackage Twenty_Seventeen
+ * @since 1.0
+ */
+
+/**
+ * Add SVG definitions to the footer.
+ */
+function twentyseventeen_include_svg_icons() {
+
+	// Define SVG sprite file.
+	$svg_icons = get_template_directory() . '/assets/images/svg-icons.svg';
+
+	// If it exists, include it.
+	if ( file_exists( $svg_icons ) ) {
+		echo file_get_contents( $svg_icons );
+	}
+
+}
+add_action( 'wp_footer', 'twentyseventeen_include_svg_icons', 9999 );
+
+/**
+ * Return SVG markup.
+ *
+ * @param  array  $args {
+ *     Parameters needed to display an SVG.
+ *
+ *     @param string $icon Required. Use the icon filename, e.g. "facebook-square".
+ *     @param string $title Optional. SVG title, e.g. "Facebook".
+ *     @param string $desc Optional. SVG description, e.g. "Share this post on Facebook".
+ * }
+ * @return string SVG markup.
+ */
+function twentyseventeen_get_svg( $args = array() ) {
+
+	// Make sure $args are an array.
+	if ( empty( $args ) ) {
+		return esc_html__( 'Please define default parameters in the form of an array.', 'twentyseventeen' );
+	}
+
+	// Define an icon.
+	if ( false === array_key_exists( 'icon', $args ) ) {
+		return esc_html__( 'Please define an SVG icon filename.', 'twentyseventeen' );
+	}
+
+	// Set defaults.
+	$defaults = array(
+		'icon'        => '',
+		'title'       => '',
+		'desc'        => '',
+		'aria_hidden' => true, // Hide from screen readers.
+	);
+
+	// Parse args.
+	$args = wp_parse_args( $args, $defaults );
+
+	// Set aria hidden.
+	$aria_hidden = '';
+
+	if ( true === $args['aria_hidden'] ) {
```

Comment:
Right, but I'm suggesting that in the code here you set `aria_hidden` based on whether there's a title/desc -- my thinking is that any time someone sets a title or desc, they will want `aria_hidden` as false, so why not save them a step? And likewise, you want `aria_hidden` to be true if there is no title/desc, because then there's nothing for a screen reader to read.